### PR TITLE
Code maintenance/50748 remove obsolete file storages translations - Automation via `i18n-tasks` gem

### PR DESCRIPTION
**Pull request by akabiru**
### https://community.openproject.org/work_packages/50748

Integrate [`i18n-tasks`](https://github.com/glebm/i18n-tasks) helps find and manage missing and unused translations. We only begin with `modules/storages/config/locales/en.yml` pending dev discussions on whether this is a useful tool to roll out across the project

#### Before

Screenshot 2023-12-13 at 7 28 35 PM

#### After

Screenshot 2023-12-13 at 7 37 58 PM

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -253,10 +253,10 @@ group :test do
   gem 'rails-controller-testing', '~> 1.0.2'
 
   gem 'capybara', '~> 3.39.0'
+  gem 'capybara_accessible_selectors', git: 'https://github.com/citizensadvice/capybara_accessible_selectors', branch: 'main'
   gem 'capybara-screenshot', '~> 1.0.17'
   gem 'cuprite', '~> 0.15.0'
   gem 'selenium-webdriver', '~> 4.16.0'
-  gem 'capybara_accessible_selectors', git: 'https://github.com/citizensadvice/capybara_accessible_selectors', branch: 'main'
```
Comment (aaron-contreras) [inline Gemfile]: 👍

File: spec/i18n_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+require 'i18n/tasks'
+
+RSpec.describe I18n do
+  let(:i18n) { I18n::Tasks::BaseTask.new }
+  let(:missing_keys) { i18n.missing_keys }
+  let(:unused_keys) { i18n.unused_keys }
+  let(:inconsistent_interpolations) { i18n.inconsistent_interpolations }
+
+  it 'does not have missing keys' do
+    skip 'Enable when i18n-tasks is enabled across the project, otherwise this spec will report false positives'
```
Comment (aaron-contreras) [inline spec/i18n_spec.rb]: I always prefer `pending` with these kinds of to-do specs. We'll know it's fixed once it starts reporting as a failure

Comment (akabiru) [reply spec/i18n_spec.rb]: Good call 👍🏾

**Review by oliverguenther — APPROVED** [review]
I share the concerns of @klaustopher who stated that this breaks apart fast if you have a lot of dynamic strings, but I'd be fine with slowly rolling it out. 

If we do merge this, could you create a followup code maintenance ticket to extends the covered yml files?

**Review by ba1ash — APPROVED** [review]
LGTM :+1:

**Comment by akabiru** [comment]
> I share the concerns of @klaustopher who stated that this breaks apart fast if you have a lot of dynamic strings, but I'd be fine with slowly rolling it out.
> 
> If we do merge this, could you create a followup code maintenance ticket to extends the covered yml files?

Sure, I think the main benefit here is more of a convenience tool for cleaning up statically defined translations that I reckon form the majority of definitions. It implicitly discourages use of dynamic definitions.

---

_WP created for incremental rollout https://community.openproject.org/wp/51762_

**Pull request merged**